DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,178 @@
+- [Developer Guide](#developer-guide)
+    - [Getting Started](#getting-started)
+        - [Git Clone logstash-output-opensearch](#git-clone-logstash-output-opensearch-repo)
+        - [Install Prerequisites](#install-prerequisites)
+            - [JRuby](#JRuby)
+            - [Docker](#docker)
+        - [Run Tests](#run-tests)
+        - [Run plugin](#run-plugin-in-logstash)
+        - [Configuration for Logstash Output OpenSearch Plugin](#configuration-for-logstash-output-opensearch-plugin)
+    - [Submitting Changes](#submitting-changes)
+
+# Developer Guide
+
+So you want to contribute code to logstash-output-opensearch? Excellent! We're glad you're here. Here's what you need to do.
+
+## Getting Started
+
+### Git Clone logstash-output-opensearch Repo
+
+Fork [opensearch-project/logstash-output-opensearch](https://github.com/opensearch-project/logstash-output-opensearch) and clone locally, e.g. `git clone https://github.com/[your username]/logstash-output-opensearch.git`.
+
+### Install Prerequisites
+
+#### JRuby
+
+This plugin builds using JRuby. This means you'll need JRuby with the Bundler gem installed.
+
+One easy way to get JRuby on is to use [rvm](https://rvm.io/rvm). Please check [here](https://rvm.io/rvm/install) on how to install rvm on your system. 
+If you use [rvm](https://rvm.io/rvm), then installing JRuby is also a piece of cake:
+```bash
+#Installing jruby
+rvm install jruby
+
+#Installing bundler
+jruby -S gem install bundler 
+```
+
+#### Docker
+
+[Docker](https://docs.docker.com/install/) is required for executing unit and integration tests.
+
+### Run Tests
+
+We run both unit tests and integration tests inside docker environment. 
+Perform the following from your project root directory eg: `~/workspace/logstash-output-opensearch`
+
+1. Tests against OpenSearch clusters.
+
+```bash
+# Set up Environment variable for docker to pull your test environment
+export LOGSTASH_VERSION=7.13.2 # mandatory 
```

Comment:
Can this be defaulted. I ran into https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/1019 which seems more annoying than anything else.

Reply:
Ack.